kmod/src/xattr.c

```diff
@@ -740,6 +811,17 @@ static int scoutfs_xattr_set(struct dentry *dentry, const char *name,
 		undo_totl = true;
 	}
 
+	if (tgs.worm) {
+		ret = parse_worm_name(name, name_len);
+		if (ret < 0)
+			 goto release;
+		else {
+			ret = parse_worm_timeval(si, value, size);
```

Comment:
This doesn't crash on deletion (value = NULL and size = 0) 'cause the parsing loop saw size = 0.   Don't put it here, move the parse_worm_timeval down into the rest of the value parsing in if (value).   You can see how if (tgs.totl) does it and add this just after it.